### ISSUE TITLE
ExecutionEngineException when overriding with generic type parameter (covariant return types)

### ISSUE THREAD

**Issue by Phantonia** [dotnet/runtime#46696]
### Description

The following code compiles successfully with C# 9, however, at runtime it throws an ExecutionEngineException:
```cs
new Derived<Cat>();

class Base
{
    public virtual Animal Animal { get; }
}

class Derived<T> : Base where T : Animal
{
    public override T Animal { get; }
}

class Animal { }

class Cat : Animal { }
```

The code is using C# 9's covariant return types to override the virtual `Animal` property of `Base`. The type is uses is the type parameter `T`. This is typesafe because the type parameter is constrained to `Animal`.

It only throws when an instance of `Derived<T>` is created or a static method is called.

### Configuration

.NET version: 5.0
OS: Windows 10 Home
OS version: 2004
Architecture: x64

**Comment by janvorli** [comment]
This is a dup of #45037. It was already fixed in #45275 and the fix is part of the upcoming 5.0.2 release.

**Comment by Phantonia** [comment]
Ah, thank you ^^
